pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -124,6 +127,8 @@ type pausedVMIs struct {
 	paused map[types.UID]bool
 }
 
+var loopbackAddress = ip.GetLoopbackAddress()
```

Comment:
Executing io actions at pre-main is inappropriate and having such a global is a high risk (like any global).
This may also be logically incorrect as the state of the pod when the program starts may not be the same later on during its lifetime.

There are a few options here, usually the nicer ones are also more complex. The inability to stub it easily is a result of sub-optimal design of the current code.
Here we go:
- Global function pointer at the (new) `ip` package: A test should be able to overwrite it by placing a stub.
  I do not like it much, but it is much better than the current option.
- Inject a `IpInfo` handler/object into the `LibvirtDomainManager`, used to fetch IP information during its operation. The loopback function can be placed in it.
- The previous options may come with an interface of `ip`, such that it can be substituted in the test.

Note: There may be other options but I found little on the subject in general.

We can go with the first, reasoning that the current code is too messy to adjust to fit the other options. But this is a smell that things are not in a good shape at the moment.

Reply:
> This may also be logically incorrect as the state of the pod when the program starts may not be the same later on during its lifetime.

how does the loopback address changes during the lifetime of the pod? I don't see how this is a high risk

Reply:
Yes, executing this here is probably not a good idea, but this variable can be populated from somewhere else.
I don't this we should overengineer this. It looks very straight forward and this address shouldn't change during the lifetime of the pod - I don't see a reason why would it.

Reply:
Enabling and disabling IPv6 may be actively changed during the lifetime of the pod.
The inability to run parallel tests is also a concern and a limitation of globals.
Globals look innocent and simple solutions at first, then they mess up and slow down progress in various ways. Please strive not to use them and when you do, feel bad about it (i.e. feel the urge to refactor).

Reply:
I vote as well not to over-engineer please, i did in this PR enough out of the box.
PR scopes should be limited in their scope.
Its not that i violated a rule, and cant block this PR for inventing new ways.
We should consider that soon there will be dual-stack and the code might going to change and dictates new requirements,
We are the ones that control the pod settings.
For now we there isn't in the agenda a dynamic pod changes.
Adding more and more code without a feature need and a way to test it naturally in a real test case, isn't the right way imo.
We should focus on creating value, each one see value in a different way.

Reply:
I disagree and feel uncomfortable that I need to argue against global/s repeatedly.
I can think of several globals that make sense to me, this is not one of them.

Reply:
I wrote more reasons beside the issue if its global or not.
I am retesting and falling about blockers again and again because the PR is stalling,
each iterations with new comments, some even from the previous iterations,
please consider that there are ways to be more efficient for me, and not just for me.
we can take it into another PR, in a fresh clean way please if needed.
On this PR i ask and vote to wrap up please.

Reply:
There is a bug now in ipv4 that i must first take care of, those iterations cost a lot, we did around 3 variations of the code. Its more than enough.

Reply:
what about adding changing the condition to the function that only if
KUBERNETES_SERVICE_HOST has an IPv6 value
then use IPv6 address ?

This value exists in the pod env, as mandatory for K8s pods
and is even better, because it will handle case of IPv4 mapping disabled
And will work for both the unit test and the official code

Reply:
Reading the capabilities from the stack is the correct  approach for the production code.
You can use such techniques for tests, but I do not think it will help here.

Reply:
We will have a problem when IPv4 mapping is disable, then we will must need to determine what IP to use, according the pod network, this is outside of this PR scope.
But we could use this in order to reduce the need of stubbing.

Anyway i will first work on making this work again, something broken lately.
And then refocus on this.